messenger/subscription.go

```diff
@@ -0,0 +1,117 @@
+package messenger
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/status-im/status-go/geth/node"
+)
+
+// MsgHandler is a callback function that processes messages delivered to
+// asynchronous subscribers.
+type MsgHandler func(msg *Msg)
+
+// Subscription : allows you manage subscriptions to a specific topic
+type Subscription struct {
+	filters     []string
+	handler     MsgHandler
+	sn          *node.StatusNode
+	unsubscribe chan bool
+}
+
+// NewSubscription : creates a new subsription object
+func NewSubscription(sn *node.StatusNode, filters []string, handler MsgHandler) *Subscription {
+	return &Subscription{
+		sn:      sn,
+		filters: filters,
+		handler: handler,
+	}
+}
+
+// Subscribe : Listens to a specific topic and executes the given function for
```

Comment:
@adriacidre There's no "given" function here (nothing is passed as an argument to `Subscribe`)...